StrongMind/canvas_shim
# Qty 1362

[QTY-1362](https://strongmind.atlassian.net/browse/QTY-1362)

## Purpose 
Ensure completion requirements are reflected in the UI appropriately based on the passing threshold in settings.

## Approach 
Modify requirements_service and context_module_importer to add the correct min_score, must_view, must_submit, or must_contribute completion requirement based on the assignment group, assignment type, and passing threshold.

## Testing
Tested locally and on newidsandbox. To confirm thresholds are set and updated as expected some example steps might be:

1. Set thresholds on account settings page
2. Create a new course and import a cartridge, verify that the thresholds set are applied to the course settings page and assignments based on the content's assignment group
3. Edit an assignments assignment group, verify that the threshold changed as appropriate
4. Edit the course level thresholds, verify that content within the course has threshold updated as appropriate
5. Manually create an assignment and a discussion topic (since they are created differently) and verify that both have the appropriate thresholds set (DiscussionTopic will not have a threshold set until marked as graded as that is when it has an associated assignment created and therefor an assignment group threshold to associate)
6. Create courses with a start date in the past and one in the future
7. Change the thresholds set in account settings
8. Verify that the course with a past start date is not affected but the course with the future start date reflects the new thresholds
9. Verify that courses which have passing thresholds overridden are not affected by the change at the account level but that course level thresholds which match the account level setting are updated.
10. Verify UI displays view, submit, or contribute appropriately based on completion requirement. 

## Screenshots/Video
![image](https://user-images.githubusercontent.com/119442731/211422440-25942f14-5ffb-40de-a2a5-f34c8494a4ee.png)


[QTY-1362]: https://strongmind.atlassian.net/browse/QTY-1362?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
